subsystems/viamserver/viamserver.go

```diff
@@ -294,17 +296,79 @@ func (s *viamServer) HealthCheck(ctx context.Context) (errRet error) {
 	return errRet
 }
 
+func (s *viamServer) isRestartAllowed(ctx context.Context) (restartAllowed bool, errRet error) {
+	for _, url := range []string{s.checkURL, s.checkURLAlt} {
+		s.logger.Debugf("starting restart allowed check for %s using %s", SubsysName, url)
+
+		ctx, cancelFunc := context.WithTimeout(ctx, time.Second*10)
+		defer cancelFunc()
+
+		url += "/restart_status"
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+		if err != nil {
+			return false, errw.Wrapf(err, "checking whether %s allows restart", SubsysName)
+		}
+
+		// disabling the cert verification because it doesn't work in offline mode (when connecting to localhost)
+		//nolint:gosec
+		client := &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return false, errw.Wrapf(err, "checking whether %s allows restart", SubsysName)
+		}
+
+		defer func() {
+			errRet = errors.Join(errRet, resp.Body.Close())
```

Comment:
I know this is a pattern in our codebase, but (I claim) there's no utility in checking the error on a `Close`.

If the whole function worked fine, should a caller not proceed because `Close` was unhappy?

Reply:
I would agree, was just matching pattern from `HealthCheck` above. Used `utils.UncheckedError` instead here and above, but I don't have a strong preference. Either way, it would be good if the HTTP response `Close` here and above handled their errors identically.